lib/ansible/galaxy/collection/__init__.py

```diff
@@ -1580,6 +1602,27 @@ def _resolve_depenency_map(
         include_signatures,  # type: bool
 ):  # type: (...) -> dict[str, Candidate]
     """Return the resolved dependency map."""
+    if not HAS_RESOLVELIB:
+        raise AnsibleError("Failed to import resolvelib, check that a supported version is installed")
+    if not HAS_PACKAGING:
+        raise AnsibleError("Failed to import packaging, check that a supported version is installed")
+    try:
+        dist = distribution('ansible-core')
+    except PackageNotFoundError:
```

Comment:
I spotted an issue with this. This should probably be `except Exception:` so req is always defined before `finally` runs. Just in case there is an unexpected error with calling distribution(). I'm going to open a PR for devel and cherry-pick to the commit to this PR before it should be merged.

Reply:
Fix: https://github.com/ansible/ansible/pull/77993